install/kubernetes/cilium/templates/cilium-nodeinit-daemonset.yaml

```diff
@@ -61,8 +61,39 @@ spec:
           imagePullPolicy: {{ .Values.nodeinit.image.pullPolicy }}
           securityContext:
             privileged: true
-{{- if .Values.nodeinit.revertReconfigureKubelet }}
           lifecycle:
+{{- if and (.Values.eni.enabled) (not .Values.enableIPv4Masquerade) }}
+            postStart:
+              exec:
+                command:
+                  - "nsenter"
+                  - "-t"
+                  - "1"
+                  - "-m"
+                  - "--"
+                  - "/bin/sh"
+                  - "-c"
+                  - |
+                    #!/bin/bash
+
+                    set -o errexit
+                    set -o pipefail
+                    set -o nounset
+
+                    # When running in AWS ENI mode, it's likely that 'aws-node' has
+                    # had a chance to install SNAT iptables rules. These can result
+                    # in dropped traffic, so we should attempt to remove them.
+                    # We do it using a 'postStart' hook since this may need to run
+                    # for nodes which might have already been init'ed but may still
+                    # have dangling rules. This is safe because there are no
+                    # dependencies on anything that is part of the startup script
+                    # itself, and can be safely run multiple times per node (e.g. in
+                    # case of a restart).
+                    echo 'Deleting iptables rules created by the AWS CNI VPC plugin'
+                    iptables-save | grep -v AWS-SNAT-CHAIN | iptables-restore
```

Comment:
I would be better to check if there are any `AWS-SNAT-CHAIN` entries, and avoid doing anything if none are present.

Reply:
@errordeveloper makes sense — I've amended the PR, could you have another look please?

Reply:
Thanks, updated version looks better!